pkg/status/status.go

```diff
@@ -155,6 +160,21 @@ func NewProber(
 	}
 }
 
+func (m *Prober) IsProbeActive(ing *v1alpha1.Ingress) (ProbeState, bool) {
+	ingressKey := types.NamespacedName{Namespace: ing.Namespace, Name: ing.Name}
+
+	state, ok := func() (ProbeState, bool) {
```

Comment:
Is there a reason (that I'm not seeing) that this is wrapped in a immediately executed inline function?

Reply:
No I can dropped this - I copied it from the earlier IsReady that did extra work so it didn't want to hold the lock